src/settings/accounts-settings.test.tsx

```diff
@@ -0,0 +1,756 @@
+			}
+		];
+
+		const account: Account = {
+			name: defaultEmail,
+			rights: { targets: [] },
+			signatures: { signature: [] },
+			id: defaultId,
+			displayName: '',
+			identities: {
+				identity: identitiesArray
+			}
+		};
+
+		const identitiesDefault = map(identitiesArray, (item, index) =>
+			identityToIdentityProps(item, index)
+		);
+
+		identitiesDefault.unshift(identitiesDefault.pop() as IdentityProps);
+
+		const { user } = setup(
+			<AccountsSettings account={account} identitiesDefault={identitiesDefault} />
+		);
+		await screen.findByText('sendAs');
+		expect(screen.getByText(persona1FullName)).toBeVisible();
+
+		await user.click(screen.getByRole('button', { name: /add persona/i }));
+		expect(screen.getByText('New Persona 1')).toBeVisible();
+	});
+
+	test('Should remove the identity from the list on delete action', async () => {
+		const defaultFirstName = faker.person.firstName();
+		const defaultLastName = faker.person.lastName();
+		const defaultFullName = faker.person.fullName({
+			firstName: defaultFirstName,
+			lastName: defaultLastName
+		});
+		const defaultEmail = faker.internet.email({
+			firstName: defaultFirstName,
+			lastName: defaultLastName
+		});
+		const defaultId = faker.string.uuid();
+
+		const persona1FullName = faker.person.fullName();
+		const persona1Email = faker.internet.email();
+		const persona1Id = faker.string.uuid();
+
+		const identitiesArray: Account['identities']['identity'] = [
+			{
+				id: persona1Id,
+				name: persona1FullName,
+				_attrs: {
+					zimbraPrefIdentityName: persona1FullName,
+					zimbraPrefReplyToEnabled: 'FALSE',
+					zimbraPrefFromDisplay: '',
+					zimbraPrefFromAddress: persona1Email,
+					zimbraPrefIdentityId: persona1Id
+				}
+			},
+			{
+				id: defaultId,
+				name: 'DEFAULT',
+				_attrs: {
+					zimbraPrefIdentityName: defaultFullName,
+					zimbraPrefReplyToEnabled: 'FALSE',
+					zimbraPrefFromDisplay: defaultFirstName,
+					zimbraPrefFromAddress: defaultEmail,
+					zimbraPrefIdentityId: defaultId
+				}
+			}
+		];
+
+		const account: Account = {
+			name: defaultEmail,
+			rights: { targets: [] },
+			signatures: { signature: [] },
+			id: defaultId,
+			displayName: '',
+			identities: {
+				identity: identitiesArray
+			}
+		};
+
+		const identitiesDefault = map(identitiesArray, (item, index) =>
+			identityToIdentityProps(item, index)
+		);
+
+		identitiesDefault.unshift(identitiesDefault.pop() as IdentityProps);
+
+		const { user } = setup(
+			<AccountsSettings account={account} identitiesDefault={identitiesDefault} />
+		);
+		await screen.findByText('sendAs');
+		const persona1Row = screen.getByText(persona1FullName);
+		expect(persona1Row).toBeVisible();
+		await user.click(persona1Row);
+		await user.click(screen.getByRole('button', { name: /delete/i }));
+		const confirmButton = await screen.findByRole('button', { name: /delete permanently/i });
+		await user.click(confirmButton);
+		expect(persona1Row).not.toBeInTheDocument();
```

Comment:
There is a missing await here which is causing the test suites to fail randomly based on timers.
When you delete the persona1, the default identity will be then selected, causing a second render of the delegates section, and so the effect run, the async network request is run etc..
```suggestion
        await screen.findBytext('sendAs');
		expect(persona1Row).not.toBeInTheDocument();
```